src/dbnode/storage/database.go

```diff
@@ -340,6 +358,32 @@ func (d *db) addNamespacesWithLock(namespaces []namespace.Metadata) error {
 	return nil
 }
 
+func (d *db) updateNamespaceSchemasWithLock(schemaUpdates []namespace.Metadata) error {
+	for _, n := range schemaUpdates {
+		// ensure namespace exists
+		existing, ok := d.namespaces.Get(n.ID())
+		if !ok { // should never happen
+			return fmt.Errorf("non-existing namespace marked for schema update: %v", n.ID().String())
+		}
+		currentId := "none"
+		current, found := existing.SchemaRegistry().GetLatest()
+		if found {
+			currentId = current.DeployId()
+		}
+		// log schema update
+		latest, found := n.Options().SchemaRegistry().GetLatest()
+		if !found {
+			return fmt.Errorf("failed to update latest schema for namespace %s", n.ID().String())
+		}
+		d.log.Infof("updating database namespace (%s) schema from %s to %s", n.ID().String(), currentId, latest.DeployId())
```

Comment:
Ah I see you have a log about the schema change here. Could we add the `oldSchema` and the `newSchema` as string fields on the log message to aid debugging and operations?

Reply:
Also I think it might be preferable if this log was emitted after the error check below so users never see it if the update is gonna fail when you call `SetSchemaRegistry`